static/js/publisher/metrics/graphs/activeDevices.js

```diff
@@ -28,66 +29,48 @@ function findEnd(data) {
   return data.length - 1;
 }
 
-function getHighest(data) {
-  let highestIndex = 0;
-  let highestValue = 0;
-  for (let i = 0, ii = data.length ; i < ii; i += 1) {
-    if (data[i] > highestValue) {
-      highestValue = data[i];
-      highestIndex = i;
-    }
-  }
-
-  return {
-    index: highestIndex,
-    value: highestValue
-  };
-}
+const labelPosition = function(graphEl, labelText, points) {
+  const start = findStart(points);
+  const end = findEnd(points);
 
-const labelPosition = function(graphEl, labelText, data, isFirst, isLast) {
-  const start = findStart(data);
-  const end = findEnd(data);
-  const highestIndex = getHighest(data).index;
-
-  const labelClass = labelText.replace('.', '-');
+  const labelClass = labelText.replace(/\W+/g, '-');
   const area = graphEl.querySelector(`.bb-areas-${labelClass}`);
   const bbox = area.getBBox();
-  const sliceWidth = Math.round(bbox.width / data.length);
 
-  const width = sliceWidth * (end - start);
-  let leftPadding = (width / 12);
-  
-  if (isLast && highestIndex > end - 3) {
-    leftPadding = -(width / 12);
-  }
+  const pointsWidth = (bbox.width / points.length) * (end - start);
 
-  const highestLeft = parseInt(
-    graphEl.querySelector(`.bb-event-rect-${highestIndex}`).getAttribute('x'),
-    10
-  );
+  if (bbox.height < 20) {
+    return;
+  }
 
   let label = document.querySelector(`[data-version="${labelText}"]`);
   if (!label) {
     label = document.createElementNS('http://www.w3.org/2000/svg', 'text');
+    label.setAttribute('class', 'snapcraft-metrics__active-devices-label');
     label.setAttribute('data-version', labelText);
     label.setAttribute('stroke', '#000');
-    label.setAttribute('x', highestLeft + leftPadding);
-    label.setAttribute('y', bbox.y + (bbox.height / 2));
     const labelSpan = document.createElementNS('http://www.w3.org/2000/svg', 'tspan');
```

Comment:
Oh boy, it's a pain that we need to be that low level...
But I guess we need to manually create SVG parts here, right?

Reply:
Yeah we could add *ANOTHER* library, but it seemed pointless for such a small part of the project.